### PR TITLE
Add an overflow check in truncate implementation for Unix.

### LINKED ISSUE #63326: File::set_len uses an unsafe cast on Unix
Precisely, `size as off64_t`. The conversion occurs [here](https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/fs.rs#L558) 
On my system `off64_t` is `i64`. This means that the number will overflow if larger than `i64::max_value()` and become negative. In such case, `ftruncate` will return an `EINVAL`: (from `man ftruncate`:
```
       EINVAL The argument length is negative or larger than the maximum file size.
```

Should we manually return an error should an overflow occur or is the `EINVAL` returned by `ftruncate` the intended behavior? I would return an `io::ErrorKind::InvalidInput` with the payload indicating that an overflow occured (or even just `TryFromIntError(())` which `try_into` returns)

This issue was noticed while discussing CraneStation/wasi-common#47.
cc @alexcrichton 

I will implement the fix once we've decided what kind of behavior we want.

### PR DESCRIPTION
**Pull request by marmistrz**
Closes #63326.
cc @alexcrichton

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

File: src/libstd/sys/unix/fs.rs [hunk 1]
```diff
@@ -554,9 +555,14 @@ impl File {
         return crate::sys::android::ftruncate64(self.0.raw(), size);
 
         #[cfg(not(target_os = "android"))]
-        return cvt_r(|| unsafe {
-            ftruncate64(self.0.raw(), size as off64_t)
-        }).map(|_| ());
+        {
```
Comment (lzutao) [inline src/libstd/sys/unix/fs.rs]: import `use crate::convert::TryInto;` here only.

**Review by tesuji — COMMENTED** [review]
Failed in https://github.com/rust-lang/rust/pull/63354#issuecomment-519286605

File: src/libstd/sys/unix/fs.rs (hunk 1 above)
Comment (marmistrz) [reply src/libstd/sys/unix/fs.rs]: Done

**Pull request merged**